### ISSUE TITLE
AuthService no longer recognized tls options

### ISSUE THREAD

**Issue by HT154** [datawire/ambassador#1202]
**Describe the bug**
v1 AuthService throws an error when specifying the `tls` property.

```
auth-web.default.1: not a valid AuthService: Additional properties are not allowed ('tls' was unexpected) Failed validating 'additionalProperties' in schema: {'$schema': 'http://json-schema.org/schema#', 'additionalProperties': False, 'id': 'https://getambassador.io/schemas/mapping.json', 'properties': {'allow_request_body': {'type': 'boolean'}, 'allowed_authorization_headers': {'items': {'type': 'string'}, 'type': 'array'}, 'allowed_request_headers': {'items': {'type': 'string'}, 'type': 'array'}, 'ambassador_id': {'anyOf': [{'type': 'string'}, {'items': {'type': 'string'}, 'type': 'array'}]}, 'apiVersion': {'enum': ['ambassador/v1']}, 'auth_service': {'type': 'string'}, 'kind': {'type': 'string'}, 'name': {'type': 'string'}, 'path_prefix': {'type': 'string'}, 'proto': {'oneOf': [{'enum': ['http']}, {'enum': ['grpc']}]}, 'timeout_ms': {'type': 'integer'}}, 'required': ['apiVersion', 'kind', 'name', 'auth_service'], 'type': 'object'} On instance: {'apiVersion': 'ambassador/v1', 'auth_service': 'auth-web:443', 'kind': 'AuthService', 'name': 'auth-service', 'tls': True}
```

**To Reproduce**
```yaml
"apiVersion": "ambassador/v1"
"auth_service": "auth-web:443"
"kind": "AuthService"
"name": "auth-service"
"tls": true
```
**Expected behavior**
The AuthService annotation continues to support the `tls` flag with the same semantics as before.

**Versions (please complete the following information):**
 - Ambassador: 0.50.0
 - Kubernetes environment: On-prem (kubeadm)
 - Version: v1.11.5

**Comment by LiVe** [comment]
we're currently facing the same issue. we'd like to specify the tls context to use a service in our istio mesh to handle auth.
e.g.:
tls: upstream

**Comment by nbkrause** [comment]
It appears the `tls:` attribute was dropped from the v1 `AuthService` config. This should be corrected so we can support @LiVe's use case of having Istio secure the traffic between Ambassador and the external auth service.

@HT154 In the mean time, enabling https to the external auth service is still supported by specifying it in the `auth_service` attribute e.g.
```
"apiVersion": "ambassador/v1"
"auth_service": "https://auth-web:443"
"kind": "AuthService"
"name": "auth-service"
```

**Comment by richarddli** [comment]
Fixed in 0.50.2
